File: src/main/java/uk/co/alt236/apkcompare/comparators/signature/SignatureComparator.java
package uk.co.alt236.apkcompare.comparators.signature;

import uk.co.alt236.apkcompare.apk.Apk;
import uk.co.alt236.apkcompare.comparators.ApkComparator;
import uk.co.alt236.apkcompare.comparators.results.ResultSection;

import java.util.ArrayList;
import java.util.List;

public class SignatureComparator implements ApkComparator {
    private final V1SignatureComparator v1SignatureComparator;
    private final V2SignatureComparator v2SignatureComparator;

    public SignatureComparator() {
        v1SignatureComparator = new V1SignatureComparator();
        v2SignatureComparator = new V2SignatureComparator();
    }

    @Override
    public List<ResultSection> compare(Apk apk1, Apk apk2) {
        final List<ResultSection> retVal = new ArrayList<>();

        retVal.addAll(v1SignatureComparator.compare(apk1, apk2));
        retVal.addAll(v2SignatureComparator.compare(apk1, apk2));

        return retVal;
    }
}

File: app/src/main/java/uk/co/alt236/apkcompare/app/comparators/signature/SignatureComparator.kt
package uk.co.alt236.apkcompare.app.comparators.signature

import uk.co.alt236.apk.Apk
import uk.co.alt236.apkcompare.app.comparators.ApkComparator
import uk.co.alt236.apkcompare.app.comparators.results.ComparisonResult
import java.util.*

class SignatureComparator : ApkComparator {
    private val v1SignatureComparator = V1SignatureComparator()
    private val v2SignatureComparator = V2SignatureComparator()

    override val name: String = "Signature Comparator"

    override fun compare(apk1: Apk, apk2: Apk): List<ComparisonResult> {
        val retVal = ArrayList<ComparisonResult>()

        retVal.addAll(v1SignatureComparator.compare(apk1, apk2))
        retVal.addAll(v2SignatureComparator.compare(apk1, apk2))

        return Collections.unmodifiableList(retVal)
    }
}